### PR TITLE
Script to enhance Bonding test case

### PR DESCRIPTION
**Pull request by hathyaga**
The enhancement contains timeout for Bonding Peer interface
and a new failure check added when all interfaces are down

Signed-off-by: Harsha Thyagaraja <harshkid@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by hathyaga** [comment]
[Bonding_Peer_joblog.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/1001649/Bonding_Peer_joblog.txt)

File: io/net/bonding.py [hunk 1]
```diff
@@ -210,8 +242,8 @@ def bond_setup(self, arg1, arg2):
                 if process.system(cmd, shell=True, ignore_status=True) != 0:
                     self.fail("Mode %s FAIL while bonding setup" % arg2)
                 time.sleep(2)
-            cmd = "cat /proc/net/bonding/%s | grep 'Bonding Mode' |\
-                  cut -d ':' -f 2" % self.bond_name
+            cmd = "self.bond_status| grep 'Bonding Mode' |\
```
Comment (narasimhan-v) [inline io/net/bonding.py]: `cmd = "%s ....." % self.bond_status`

File: io/net/bonding.py [hunk 2]
```diff
@@ -243,9 +279,9 @@ def bond_setup(self, arg1, arg2):
                 cmd += 'echo "+%s" > %s;' % (val, self.bonding_slave_file)
             for val in self.peer_interfaces:
                 cmd += 'ifconfig %s up;' % val
-            cmd += 'ifconfig %s %s netmask %s up;sleep 5;'\
-                   % (self.bond_name,
-                      self.peer_first_ipinterface, self.net_mask[0])
+            cmd += 'ifconfig %s %s netmask %s up;sleep %s;'\
+                   % (self.bond_name, self.peer_first_ipinterface,
+                      self.net_mask[0], self.peer_wait_time)
```
Comment (narasimhan-v) [inline io/net/bonding.py]: would be better if `timeout` is used instead of `sleep`

File: io/net/bonding.py [hunk 3]
```diff
@@ -283,19 +319,28 @@ def tearDown(self):
...
             cmd = "ifconfig %s %s netmask %s up"\
                   % (val1, val2, val3)
+            for i in range(0, 600, 60):
+                if process.system(cmd, shell=True, ignore_status=True) != 0:
+                    self.log.info("Unable to bring up %s interface\
+                                   in Host machine" % val1)
+                    if 'state UP' in process.system_output("ip link \
+                            show %s" % val1, shell=True):
+                        break
+                    time.sleep(60)
```
Comment (narasimhan-v) [inline io/net/bonding.py]: I might have overlooked in previous PR. But I do not get what you are trying to do here.

File: io/net/bonding.py.data/bonding.yaml [hunk 4]
```diff
@@ -13,8 +13,11 @@ Test: !mux
         bonding_mode: "5"
     seventh:
         bonding_mode: "6"
-host_interfaces: "enP3p1s0f0,enP3p1s0f1"
-peer_ip: "85.85.85.85"
-peer_interfaces: "eth3"
+host_interfaces: "enP2p1s0f2,enP2p1s0f3"
```
Comment (narasimhan-v) [inline io/net/bonding.py.data/bonding.yaml]: Please consider removing the interfaces and IP values from the yaml files when you send the next version for this PR

**Comment by hathyaga** [comment]
Closing this PR as a new PR has been sent
https://github.com/avocado-framework-tests/avocado-misc-tests/pull/504

**Pull request closed without merging**